include/irr/asset/IRenderpassIndependentPipeline.h

```diff
@@ -302,30 +302,32 @@ enum E_VERTEX_ATTRIBUTE_ID
 template<typename SpecShaderType, typename LayoutType>
 class IRenderpassIndependentPipeline : public IPipeline<LayoutType>
 {
-protected:
+public:
     _IRR_STATIC_INLINE_CONSTEXPR size_t SHADER_STAGE_COUNT = 5u;
 
-    enum {
-        VERTEX_SHADER_IX = 0,
-        TESS_CTRL_SHADER_IX = 1,
-        TESS_EVAL_SHADER_IX = 2,
-        GEOMETRY_SHADER_IX = 3,
-        FRAGMENT_SHADER_IX = 4
+    enum E_SHADER_STAGE_IX
+    {
+        ESSI_VERTEX_SHADER_IX = 0,
+        ESSI_TESS_CTRL_SHADER_IX = 1,
+        ESSI_TESS_EVAL_SHADER_IX = 2,
+        ESSI_GEOMETRY_SHADER_IX = 3,
+        ESSI_FRAGMENT_SHADER_IX = 4
     };
 
+protected:
     IRenderpassIndependentPipeline(
-        core::smart_refctd_ptr<LayoutType> _layout,
-        core::smart_refctd_ptr<SpecShaderType> _vs,
-        core::smart_refctd_ptr<SpecShaderType> _tcs,
-        core::smart_refctd_ptr<SpecShaderType> _tes,
-        core::smart_refctd_ptr<SpecShaderType> _gs,
-        core::smart_refctd_ptr<SpecShaderType> _fs,
+        core::smart_refctd_ptr<LayoutType>&& _layout,
```

Comment:
r-value array or iterator pair to move from would have been saner

Reply:
resolved

Reply:
resolved again